arangod/Aql/AqlCallList.cpp

```diff
@@ -121,9 +115,13 @@ auto AqlCallList::fromVelocyPack(VPackSlice slice) -> ResultT<AqlCallList> {
                       slice.typeName());
   }
 
-  auto expectedPropertiesFound = std::map<std::string_view, bool>{};
-  expectedPropertiesFound.emplace(StaticStrings::AqlCallListSpecific, false);
-  expectedPropertiesFound.emplace(StaticStrings::AqlCallListDefault, false);
+  // we only have 2 different keys to check. using an std::map requires
+  // dynamic memory allocation and would be wasteful. instead, use a simple
+  // std::array here to get rid of any allocations
+  auto expectedPropertiesFound =
+      std::array<std::pair<std::string_view, bool>, 2>{
```

Comment:
Why do we need bool is it always false?

Reply:
The boolean values in the array are `false` only initially. They can be updated to `true` in line 191.

Reply:
Understood